cmd/server/main_test.go

```diff
@@ -29,3 +39,71 @@ func Test_validateWebhook(t *testing.T) {
 		})
 	}
 }
+
+func Test_reloadHandler(t *testing.T) {
+
+	// Setup variables
+	httpClient := http.DefaultClient
+	validateURL := false
+	logger := log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+	tc := PromTeamsConfig{
+		Connectors: []map[string]string{
+			{
+				"connector1": "https://localhost:8080",
+			},
+		},
+		ConnectorsWithCustomTemplates: []ConnectorWithCustomTemplate{},
+	}
+
+	// Create the initial routes for our server
+	routes, err := setupServices(tc, nil, httpClient, &validateURL, logger)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Configure the reload handler
+	handler := transport.NewServer(logger, routes, nil)
+	handler.POST("/reload", func(c echo.Context) error {
+		routes, err = setupServices(tc, nil, httpClient, &validateURL, logger)
+		if err != nil {
+			logger.Log("err", err)
+			return c.JSON(500, err)
+		}
+		transport.ReloadRoutes(handler, routes, logger)
+		return c.JSON(200, tc.Connectors)
+	})
+
+	// Call GET on /connector2 and check that we get a 404 not found. 
+	// Should return 404 not found because we haven't created it yet
+	resp, body := testURL(handler, "/connector2", http.MethodGet, nil)
+	assert.Equal(t, "{\"message\":\"Not Found\"}\n", string(body))
+	assert.Equal(t, http.StatusNotFound, resp.StatusCode)
+
+	// Now add connector2 to the map array and reload the services by calling POST on /reload
+	// Should return 200 OK 
+	tc.Connectors = append(tc.Connectors, map[string]string{
+		"connector2": "https://localhost:8443",
+	})
+	resp, body = testURL(handler, "/reload", http.MethodPost, nil)
+	assert.Equal(t, "[{\"connector1\":\"https://localhost:8080\"},{\"connector2\":\"https://localhost:8443\"}]\n", string(body))
+	assert.Equal(t, http.StatusOK, resp.StatusCode)
+
+	// The reload handler should now have reloaded all routes and we should be able to call GET on /connector2
+	// We expect to get 405 method not allowed since the handler expects POST. But 405 tells us that the connector now exists in the server
```

Comment:
this can be improve by calling the `/config` GET endpoint.

Reply:
I added an additional check that calls `GET` on `/config`.